### ISSUE TITLE
After upgrading to 1.0.11 (Homebrew) now gives error message whenever trying to use keybase

### ISSUE THREAD

**Issue by jlb1234** [keybase/keybase-issues#2074]
I upgraded to the latest version (1.0.11) via brew but now I'm getting the following error message whenever trying to run a command, such as:

keybase id max
▶ WARNING Restarting after upgrade; service is running v1.0.9-0, while v1.0.11-0 is available
▶ WARNING Failed to register as a logger: EOF
▶ ERROR EOF

**Comment by maxtaco** [comment]
Quick fix via 'pkill keybase' but I will look further into it. Sigh I
thought I had fixed this bug!!!!

**Comment by maxtaco** [comment]
Also, if you can do a `keybase log send`, I'd like to see what's going on.  I unfortunately can't repro this bug!

**Comment by jlb1234** [comment]
Thanks Max - I just tried to `pkill keybase` and it doesn't seem to have done much...

`keybase log send`
▶ WARNING Restarting after upgrade; service is running v1.0.9-0, while v1.0.11-0 is available
▶ WARNING Failed to register as a logger: EOF
This command will send recent keybase log entries to keybase.io
for debugging purposes only.

These logs don’t include your private keys or encrypted data,
but they will include filenames and other metadata keybase normally
can’t read, for debugging purposes.

Continue sending logs to keybase.io? (type 'YES' to confirm): YES
▶ ERROR EOF

`pkill keybase`

`keybase status`
▶ WARNING Restarting after upgrade; service is running v1.0.9-0, while v1.0.11-0 is available

We found a Keybase service (homebrew.mxcl.keybase) but it's not running.
You might try starting it: keybase launchd start homebrew.mxcl.keybase

▶ ERROR dial unix /Users/jason/Library/Caches/Keybase/keybased.sock: connect: no such file or directory

`keybase launchd start homebrew.mxcl.keybase`
▶ INFO Starting homebrew.mxcl.keybase

`keybase status`
▶ WARNING Restarting after upgrade; service is running v1.0.9-0, while v1.0.11-0 is available
▶ WARNING Failed to register as a logger: EOF
▶ ERROR EOF

`keybase log send`
▶ WARNING Restarting after upgrade; service is running v1.0.9-0, while v1.0.11-0 is available

We found a Keybase service (homebrew.mxcl.keybase) but it's not running.
You might try starting it: keybase launchd start homebrew.mxcl.keybase

▶ ERROR dial unix /Users/jason/Library/Caches/Keybase/keybased.sock: connect: no such file or directory

**Comment by maxtaco** [comment]
Ok, thanks for trying this out. @gabriel do you have any suggestions here?

**Comment by maxtaco** [comment]
(It might also be useful to see the end of your service log:

```
tail -1000 /Users/jason/Library/Logs/homebrew.mxcl.keybase.log
```

)

**Comment by ChrisBuchholz** [comment]
This is happening to me too. I tried uninstalling and installing keybase again, via homebrew, and now I'm getting this error:

``ERROR dial unix /Users/cb/Library/Caches/Keybase/keybased.sock: connect: no such file or directory```

**Comment by maxtaco** [comment]
@ChrisBuchholz if you can send me a log as above, that would be helpful.  I guess `log send` won't work, but tailing the end of your brew log will help me figure out what's going on. Thanks..  

Also, you might try `keybase launchd start homebrew.mxcl.keybase` as described above. Thanks

**Comment by ChrisBuchholz** [comment]
Here is some output from the keybase log. I guess that should be of help to you: https://gist.github.com/ChrisBuchholz/c59c9f728d999a2c642d

(re)starting the service doesn't work btw.:
```
$ keybase launchd start homebrew.mxcl.keybase
▶ INFO Starting homebrew.mxcl.keybase
$ keybase log send
▶ ERROR dial unix /Users/cb/Library/Caches/Keybase/keybased.sock: connect: no such file or directory
```

**Comment by maxtaco** [comment]
Are those log messages recent or just very old?  My guess is that's from an old service that's no longer running, since I believe we added a date to the log output.  Can't tell since it's just a time, unfortunately.

Can you also try a `keybase -d status`?  I'm sorry to be asking you lots of questions, but I can't seem to repro this issue.

Also, can you `cat ~/Library/LaunchAgents/homebrew.mxcl.keybase.plist`? 

thanks!

**Comment by jlb1234** [comment]
keybase log: https://gist.github.com/jlb1234/d0b4d58c72a975903a3e <https://gist.github.com/jlb1234/d0b4d58c72a975903a3e>
keybase -d status: https://gist.github.com/jlb1234/20a59952e1b2d50af058 <https://gist.github.com/jlb1234/20a59952e1b2d50af058>
keybase plist: https://gist.github.com/jlb1234/992a4a470fd284f3d767 <https://gist.github.com/jlb1234/992a4a470fd284f3d767>

**Comment by ChrisBuchholz** [comment]
@maxtaco Output from `keybase -d status`: https://gist.github.com/ChrisBuchholz/5286963385d4a5e7a4bc, and content from `~/Library/LaunchAgents/homebrew.mxcl.keybase.plist`: https://gist.github.com/ChrisBuchholz/80873e00031d67d073ba

I simply did `tail -1000 /Users/cb/Library/Logs/homebrew.mxcl.keybase.log`, just as @jib1234, to get the keybase log content, so it should be all new. The time also matches what the clock was when I did it.

**Comment by maxtaco** [comment]
I think there's a workaround over in keybase/client#2019.  Try:

```
keybase launchd unload homebrew.mxcl.keybase
```

And then try again to login / or whatever you had in mind.

We'll work on a fix for 1.0.12. Thanks for your help1

**Comment by jlb1234** [comment]
@maxtaco That seems to work

**Comment by maxtaco** [comment]
Glad it worked.  We will hopefully get a bug fix out tomorrow.  Thanks for
letting us know!

**Comment by ChrisBuchholz** [comment]
@maxtaco: `keybase launchd unload homebrew.mxcl.keybase` did not work for me, but 
```
$ launchctl unload $HOME/Library/LaunchAgents/homebrew.mxcl.keybase.plist
$ rm $HOME/Library/LaunchAgents/homebrew.mxcl.keybase.plist
$ keybase -d install
```
did, and now it seems everything is working again as it should.

**Comment by gabriel** [comment]
We tracked down the issue to an invalid launchd plist being generated if you installed version 1.0.9, and upgraded. We pushed a fix, and did a new version (1.0.12) which is now live in brew.

We missed this during our testing since it doesn't happen if you installed from and earlier version and then upgraded which is how we were testing. Thanks for helping us track down this issue.
